providers/implementations/macs/cmac_prov.c

```diff
@@ -210,6 +210,13 @@ static int cmac_set_ctx_params(void *vmacctx, const OSSL_PARAM params[])
             ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_MODE);
             return 0;
         }
+#ifdef FIPS_MODULE
+        if (EVP_CIPHER_is_a(ossl_prov_cipher_cipher(&macctx->cipher),
```

Comment:
Is it ok to just do the simple strict check here?

Reply:
I'd prefer a whitelist approach rather than blacklist.

Reply:
Will change - its a bit slower.